### ISSUE TITLE
Extern reexports don't work with `help`

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#12320]
### Describe the bug

I have a Nu file with `extern` wrappers for a CLI utility, called `apk.nu`.  It has `extern main` and `extern <subcommand>` for subcommands.  With `use apk.nu`, the completions on the command line work, but the expression `help apk add` doesn't.  Only `help add` works, which is unintuitive, and also conflicts with two other `add` subcommands which are active in my REPL.

### How to reproduce

`apk.nu`:

```Nushell
export extern main [
	--help
]

export extern add [
	...packages

	--initdb	# Initialize a new package database
	--latest (-l)	# Always choose the latest package by version
	--upgrade (-u)	# Upgrade PACKAGES and it's dependencies
	--virtual: string (-t)	# Create virtual package with given dependencies
	--no-chown	# Do not change file owner or group
]
```

REPL:

```Nushell
~/project/nu-completion/src> use apk.nu 
~/project/nu-completion/src> help apk add
Error: nu::parser::not_found

  × Not found.
   ╭─[entry #16:1:1]
 1 │ help apk add
   ·      ───┬───
   ·         ╰── did not find anything under this name
   ╰────

~/project/nu-completion/src| apk add --
--initdb   Initialize a new package database
--latest   Always choose the latest package by version
--upgrade  Upgrade PACKAGES and it's dependencies
--virtual  Create virtual package with given dependencies
```

### Expected behavior

I expected `help apk add` to show help for `apk add` extern.

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.90.1                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.76.0 (07dca489a 2024-02-04) (Alpine Linux 1.76.0-r0) |
| cargo_version      | cargo 1.76.0                                                 |
| build_time         | 2024-02-14 02:25:04 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which, zip                           |
| installed_plugins  |                                                              |

**Comment by 132ikl** [comment]
This seems to be fixed, if you still have the issue let me know and I can reopen this

**Comment by kaathewisegit** [comment]
Yep, it works correctly now!  `--no-config` works perfectly.  And in my setup the help menu shows up, but it seems that using carapace breaks something (completions don't work properly, but I'll try to isolate it some other time)
